secretsstoresynccontroller/api/v1alpha1/secretstoresync_types.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SecretObjectData defines the desired state of synchronized data within a Kubernetes secret object.
+type SecretObjectData struct {
+	// SecretDataValueSource is the data source value of the secret defined in the Secret Provider Class.
+	// +kubebuilder:validation:Required
+	SecretDataValueSource string `json:"secretDataValueSource"`
+
+	// SecretDataKey is the key in the Kubernetes secret's data field as described in the Kubernetes API reference:
+	// https://kubernetes.io/docs/reference/kubernetes-api/config-and-storage-resources/secret-v1/
+	// +kubebuilder:validation:Required
+	SecretDataKey string `json:"secretDataKey"`
```

Comment:
This should have validation to make sure that the key here is always a valid.  Grep for `func ValidateSecret(` to find the logic.